Covinator/Covinator/Pages/PfizerVaccination.cshtml.cs

```diff
@@ -21,92 +19,112 @@ public PfizerVaccination(ILogger<PfizerVaccination> logger)
+                    JSchema schema = JSchema.Parse(System.IO.File.ReadAllText("pfizerSchema.json"));
+                    JArray jsonArray = JArray.Parse(pfizerData);
+                    IList<string> validationEvents = new List<string>();
 
-                    JArray result_array = new JArray();
-                    JArray jarray = new JArray();
-                    var length_arr = jsonArray.Count();
-                    for (int i=0; i<63; i++)
-                    {
-                        result_array.Add(jsonArray[i]);
-
-                    }
+                    var currentWeek = jsonArray.FirstOrDefault()?["week_of_allocations"]?.Value<DateTime>().ToShortDateString();
 
-                    string result_string = result_array.ToString();
+                    if (jsonArray.IsValid(schema, out validationEvents))
+                    {
+                        JArray result_array = new JArray();
 
-                    var pfizerVaccineDistributionAllocations = PfizerVaccineDistributionAllocations.FromJson(result_string);
+                        for (int i = 0; i < 63; i++)
+                        {
+                            result_array.Add(jsonArray[i]);
+                        }
 
-                    /*ViewData["PfizerVaccineDistributionAllocations"] = pfizerVaccineDistributionAllocations;*/
-                    ViewData["PfizerVaccineDistributionAllocations"] = pfizerVaccineDistributionAllocations;
-                    ViewData["CurrentWeek"] = currentWeek;
+                        var pfizerVaccineDistributionAllocations = PfizerVaccineDistributionAllocations.FromJson(result_array.ToString());
 
-                }
-                else
-                {
+                        PopulateJurisdictions(jsonArray);
 
-
-                    foreach (string evt in validationEvents)
+                        ViewData["PfizerVaccineDistributionAllocations"] = pfizerVaccineDistributionAllocations;
+                        ViewData["CurrentWeek"] = currentWeek;
+                    }
+                    else
                     {
-                        Console.WriteLine(evt);
-                        ViewData["PfizerVaccineDistributionAllocations"] = new PfizerVaccineDistributionAllocations();
-                        
+                        _logger.LogError("pfizer json validation failed");
+                        foreach (string evt in validationEvents)
+                        {
+                            _logger.LogWarning($"Error while validating pfizerschema {evt}");
+                            Console.WriteLine(evt);
+                            ViewData["PfizerVaccineDistributionAllocations"] = new PfizerVaccineDistributionAllocations();
+                        }
                     }
                 }
             }
+            catch (Exception ex)
+            {
+                _logger.LogError($"Exception occured OnGet Pfizer {ex}");
+            }
```

Comment:
Implemented exception handling and logs(Warning and error levels)

Reply:
Reeeeeeeeeeeeeeeeally good.  Proper logging, instead of Console.WriteLine()